apps/files/views.py

```diff
@@ -151,3 +160,100 @@ def delete(self, request, team_slug: str, **kwargs):
     def get_success_response(self, file):
         messages.success(self.request, "File Deleted")
         return HttpResponse()
+
+
+class FileHome(LoginAndTeamRequiredMixin, TemplateView):
+    template_name = "generic/object_home.html"
+
+    def get_context_data(self, team_slug: str, **kwargs):
+        return {
+            "active_tab": "files",
+            "title": "Files",
+            "table_url": reverse("files:file_table", args=[team_slug]),
+            "enable_search": True,
+            "allow_new": False,
+        }
+
+
+class FileTableView(LoginAndTeamRequiredMixin, SingleTableView):
+    model = File
+    paginate_by = 25
+    table_class = FilesTable
+    template_name = "table/single_table.html"
+
+    def get_queryset(self):
+        collection_files_subquery = CollectionFile.objects.values_list("file_id").distinct()
+        queryset = File.objects.filter(
+            id__in=Subquery(collection_files_subquery), team=self.request.team, is_version=False
+        ).order_by("-created_at")
```

Comment:
This could get expensive - how about:

```python
is_collection_file = Exists(CollectionFile.objects.filter(file_id=OuterRef("id"))
queryset = File.objects.filter(is_collection_file).filter(is_version=False).order_by("-created_at")
```

Reply:
Thanks